tests/user/test_user_auth.py

```diff
@@ -0,0 +1,151 @@
+    authorization_encoded = base64.b64encode(credentials_encoded)
+    authorization = f"Basic {authorization_encoded.decode()}"
+    return authorization
+
+
+@pytest.fixture(autouse=True)
+def os_mock() -> Generator[Mock, None, None]:
+    with patch("personal_finances.user.user_auth.os") as mock:
+        yield mock
+
+
+@pytest.fixture(autouse=True)
+def boto3_mock() -> Generator[Mock, None, None]:
+    with patch("personal_finances.user.user_auth.boto3.client") as mock:
+        yield mock
+
+
+@pytest.fixture(autouse=True)
+def datetime_mock() -> Generator[Mock, None, None]:
+    with patch("personal_finances.user.user_auth.datetime") as mock:
+        mock.now.return_value = TEST_DATE_TIME_NOW
+        yield mock
+
+
+def _generate_test_jwt_token() -> str:
+    return jwt.encode(
+        {"exp": TEST_DATE_TIME_NOW + timedelta(hours=1), "userId": TEST_USER_ID},
+        TEST_JWT_SECRETS,
+        algorithm="HS256",
+    )
+
+
+@pytest.mark.parametrize(
+    "authorization_header, expected_response, dynamo_response_dict, env_var_dict",
+    [
+        (
+            _encode_basic_auth(TEST_USER_ID, "incorrect_password"),
+            {"statusCode": 401, "body": "User or password incorrect"},
+            TEST_USER_DYNAMO_RESPONSE,
+            TEST_ENVAR_DICT,
+        ),
+        (
+            _encode_basic_auth("incorrect_user", "incorrect_password"),
+            {"statusCode": 401, "body": "User or password incorrect"},
+            {"usernot": "found"},
+            TEST_ENVAR_DICT,
+        ),
+        (
+            "incorrect_header",
+            {"statusCode": 400, "body": "Malformed authentication header"},
+            TEST_USER_DYNAMO_RESPONSE,
+            TEST_ENVAR_DICT,
+        ),
+        (
+            _encode_basic_auth(TEST_USER_ID, TEST_USER_PASSWORD),
+            {
+                "statusCode": 200,
+                "body": json.dumps({"token": _generate_test_jwt_token()}),
+            },
+            TEST_USER_DYNAMO_RESPONSE,
+            TEST_ENVAR_DICT,
+        ),
+        (
+            _encode_basic_auth(TEST_USER_ID, TEST_USER_PASSWORD),
+            {
+                "statusCode": 500,
+                "body": "Internal Server Error",
+            },
+            Exception,
+            TEST_ENVAR_DICT,
+        ),
+        (
+            _encode_basic_auth(TEST_USER_ID, TEST_USER_PASSWORD),
+            {
+                "statusCode": 500,
+                "body": "Internal Server Error",
+            },
+            TEST_USER_DYNAMO_RESPONSE,
+            Exception,
+        ),
+    ],
+)
+def test_user_auth(
+    authorization_header: str,
+    expected_response: dict,
+    dynamo_response_dict: Union[dict, Exception],
+    env_var_dict: Union[dict, Exception],
+    os_mock: Mock,
+    boto3_mock: Mock,
+) -> None:
+    if isinstance(env_var_dict, dict):
+        os_mock.environ.__getitem__.side_effect = env_var_dict.get
+    else:
+        os_mock.environ.__getitem__.side_effect = env_var_dict
```

Comment:
`env_var_dict` name is misleading, I'd rename it to `env_var_dict_or_exception` and I'd add a comment that the `else` block is handling exceptions.